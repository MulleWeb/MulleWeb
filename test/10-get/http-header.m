#import <MulleWebClient/MulleWebClient.h>

#include <stdio.h>

static NSString   *URL = @"http://www.mulle-kybernetik.com/jagdox/dehtmlify.sh";


int  main( void)
{
   MulleCurl               *curl;
   MulleHTTPHeaderParser   *headerParser;
   NSURL                       *url;
   NSData                      *data;
   NSError                     *error;
   NSArray                     *order;
   NSDictionary                *headers;

   curl         = [[MulleCurl new] autorelease];
   headerParser = [[MulleHTTPHeaderParser new] autorelease];
   [headerParser setRecordsOrder:YES];

   [curl setDesktopTimeoutOptions];
   [curl setHeaderParser:headerParser];
//   [curl setNoBodyOptions];

   data = [curl dataWithContentsOfURLWithString:URL];
   if( ! data)
   {
      error = [NSError mulleExtract];
      fprintf( stderr, "%s\n", [[error description] UTF8String]);
      return( 1);
   }

   order   = [headerParser order];
   headers = [headerParser headers];

   printf( "%s\n", [[headerParser response] UTF8String]);
   printf( "%s\n", [[order description] UTF8String]);
   printf( "%ld headers\n", (long) [headers count]);
   printf( "%ld bytes in body\n", [data length]);

   return( 0);
}
