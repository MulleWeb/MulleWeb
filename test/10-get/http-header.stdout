HTTP/1.1 200 OK
(
    Date,
    Server,
    "Last-Modified",
    ETag,
    "Accept-Ranges",
    "Content-Length",
    "Content-Type"
)
7 headers
413 bytes in body
